### PR TITLE
Destroy remote WireGuard device on close

### PR DESCRIPTION
**Pull request by anjmao**
Disconnect was not working when using supervisor since Wgctrl close doesn't know how to remove tunnel interface.

### CODE REVIEW

File: services/wireguard/endpoint/remoteclient/client.go [hunk 1]
```diff
@@ -137,6 +142,27 @@ func (c *client) PeerStats(iface string) (*wg.Stats, error) {
...
+func (c *client) Close() (err error) {
+	c.mu.Lock()
+	defer c.mu.Unlock()
+
+	var errs []error
+	defer func() {
+		for i := range errs {
+			log.Error().Err(errs[i]).Msg("Failed to close wireguard remote client")
+			if err == nil {
+				err = errs[i]
```
Comment (Waldz) [inline services/wireguard/endpoint/remoteclient/client.go:154]: Returning just last error, `ErrorComposite.Err()` would be useful to return multiple errors

Comment (zolia) [reply services/wireguard/endpoint/remoteclient/client.go:154]: Or just wrap them all with %w

File: services/wireguard/endpoint/remoteclient/client.go [hunk 2]
```diff
@@ -137,6 +142,27 @@ func (c *client) PeerStats(iface string) (*wg.Stats, error) {
...
+
+	if err := c.DestroyDevice(c.iface); err != nil {
+		errs = append(errs, err)
+	}
+
+	if err := c.wgc.Close(); err != nil {
+		errs = append(errs, err)
+	}
+
+	return nil
```
Comment (soffokl) [inline services/wireguard/endpoint/remoteclient/client.go:167]: If you are returning `nil` here, defer will not overwrite it.

File: services/wireguard/endpoint/remoteclient/client.go [hunk 3]
```diff
@@ -137,6 +142,27 @@ func (c *client) PeerStats(iface string) (*wg.Stats, error) {
...
 }
 
-func (c *client) Close() error {
-	return c.wgc.Close()
+func (c *client) Close() (err error) {
+	c.mu.Lock()
+	defer c.mu.Unlock()
+
+	var errs []error
+	defer func() {
```
Comment (soffokl) [inline services/wireguard/endpoint/remoteclient/client.go:150]: Do you need this defer here? Just move the `for` to the end of the function.

**Pull request merged**